tests/integration/flask/test_database.py

```diff
@@ -56,6 +58,10 @@ async def test_with_database(
     await model.wait_for_idle(status=ops.ActiveStatus.name)  # type: ignore
 
     for unit_ip in await get_unit_ips(flask_app.name):
-        response = requests.get(f"http://{unit_ip}:8000/{endpoint}", timeout=5)
-        assert response.status_code == 200
+        for _ in range(10):
+            response = requests.get(f"http://{unit_ip}:8000/{endpoint}", timeout=5)
+            assert response.status_code == 200
+            if "SUCCESS" == response.text:
+                return
+            time.sleep(60)
```

Comment:
Minor nitpick: how about `await asyncio.sleep()` since this function is async?

Reply:
Updated, thanks!